tests/x11/remote_desktop/windows_server_setup.pm

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Remote Login: SLED access Windows over RDP
+# Maintainer: GraceWang <gwang@suse.com>
+# Tags: tc#1610392
+
+use strict;
+use base 'x11test';
+use testapi;
+use lockapi;
+use mmapi 'wait_for_children';
+
+sub run {
+    my $self = shift;
+
+    assert_and_click "start";
+    type_string "remote access\n";
+
+    assert_and_click "allow-remote-connections";
+    assert_screen "allow-connections-with-nla-enabled";
+
+    if (!get_var('NLA')) {
+        send_key "alt-n";
+        assert_screen "allow-connections-without-nla-enabled";
+    }
+
+    assert_and_click "system-properties-ok";
+
+    mutex_create 'win_server_ready';
+    wait_for_children;
+
+    # In case execute the client job costs too much time
+    # Add this part to deal with the lock screen
+    check_screen "grub-boot-windows";
+    if (match_has_tag 'grub-boot-windows') {
+        send_key "esc";
+        assert_screen "windows-login";
+        type_password;
+        send_key "ret";
+    }
+
+    # In case the update required window popup
+    check_screen "update-required";
+    if (match_has_tag 'update-required') {
+        assert_and_click "close-update-required";
+    }
```

Comment:
these cases of `check_screen` also could use `assert_screen` with multi-tags?

Reply:
Thanks for your comment and the offline information. 
Please let me explain why I choose to use check_screen not assert_screen with multi-tags.
There's a possibility that none of "grub-boot-windows" and "update-required" appear (if the client job finishes very quickly), so we can not use assert_screen with multi-tags since it will cause failure.
Due to the performance of the worker, the client job might cost more time than expected and usual, so either of the two screens might appear. And check_screen is a better choice for this situation.

Reply:
Yes, those two check_screen run after "wait_for_children;"
If the child job takes too much time (more than 10 mins) then the "grub-boot-windows" will appear and "update-required" might appear on the windows server, we need to deal with this situation. Otherwise, we do not need to do anything.

Reply:
Hm, I am still with mitiao here :/

Reply:
Let's suppose we are using`assert_screen [qw(grub-boot-windows update-required)];` here, if the performance of the worker is very good (I met this situation when debugging test case), and the child job finishes in 10 mins, and none of "grub-boot-windows" and "update-required" window appears, it will throw exception, then the case would be failed.

And the key test point of this case is using SLED as a client to connect to a Windows server. After verifying the client can connect to the server using remmina, the child job is finished.
As to the Windows server side, after the child job finishes, whether it automatically lock screen or not; whether the update window popup or not, IMHO that's not the key checkpoint. If they occur, I will handle this, if not, then do nothing.

Reply:
@GraceWang571 how about calling with a 600 seconds timeout? (10 minutes...)?

Reply:
I mean default timeout for automatic lock screen. But don't know why the time between win10 idle to the automatically lock screen is less than 10 mins in http://10.67.18.241/tests/959#step/windows_server_setup/6
And in the Personalisation -> Lock screen window, it shows "You need to activate Windows before you can personalise your PC".

Reply:
@mitiao @okurz @foursixnine any further comments about my last two comments?

Reply:
@GraceWang571 no further comments from me :)

Reply:
@GraceWang571 how about this http://pastebin.suse.de/25436

Reply:
@mitiao I have to admit that this is a good idea. I've updated the code and verification test run.